enumerables.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require './helpers.rb'
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_arr = []
+    my_each { |n| new_arr.push(n) if yield(n) }
+    new_arr
+  end
+
+  def my_all?(pattern = nil)
+    my_each { |n| return false unless n.to_s.match(pattern) } if pattern
+    my_each { |n| return false unless yield(n) } unless pattern
+    true
+  end
+
+  def my_any?(pattern = nil)
+    my_each { |n| return true if n.to_s.match(pattern) } if pattern
+    my_each { |n| return true if yield(n) } unless pattern
+    false
+  end
+
+  def my_none?(pattern = nil)
+    my_each { |n| return false if n.to_s.match(pattern) } if pattern
+    my_each { |n| return false if yield(n) } unless pattern
+    true
+  end
+
+  def my_count(item = false)
+    count = 0
+
+    if !block_given?
+      return length unless item
+
+      my_each { |n| count += 1 if n == item }
+    elsif block_given?
+      my_each { |n| count += 1 if yield(n) }
+    end
+    count
+  end
+
+  def my_map(proc = nil)
+    new_arr = []
+    if proc
+      my_each { |n| new_arr.push(proc.call(n)) }
+    else
+      my_each { |n| new_arr.push(yield(n)) } if block_given?
+      my_each { |n| new_arr << n } unless block_given?
+      new_arr = Enumerator.new(new_arr) unless block_given?
+      new_arr
+    end
+  end
+
+  def my_inject(*args)
+    memo = nil
+    if oop?(args[0])
+      initial = args[1]
```

Comment:
- [ ] my_inject combines all elements of enum by applying a binary operation, specified by a block
- [ ] my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method

Reply:
@samgaco 
I didn't understand this comment. I already combines the elements and uses the symbol specified when provided

Reply:
Hi @uryelah 

 I'll leave you some examples:

**When using it with a range gives a failure:**

```
irb(main):092:0> (1..2).inject(1){ |acc, n| acc + n}
=> 4
irb(main):093:0> (1..2).inject(2){ |acc, n| acc + n}
=> 5
irb(main):094:0> (1..2).inject(10){ |acc, n| acc + n}
=> 13
irb(main):095:0> (1..2).my_inject(10){ |acc, n| acc + n}
Traceback (most recent call last):
....
NoMethodError (undefined method `oop?' for 1..2:Range)
```

**also with an array**

```
irb(main):096:0> [1,2].inject(10){ |acc, n| acc + n}
=> 13
irb(main):097:0> [1,2].my_inject(10){ |acc, n| acc + n}
Traceback (most recent call last):
...
irb(main):096:0> [1,2].inject(10){ |acc, n| acc + n}
=> 13
irb(main):097:0> [1,2].my_inject(10){ |acc, n| acc + n}
Traceback (most recent call last):

```


**and for the second case same error:**

```
irb(main):098:0> [1,2].inject(2,:+)
=> 5
irb(main):099:0> [1,2].my_inject(2,:+)
Traceback (most recent call last):
....
NoMethodError (undefined method `oop?' for [1, 2]:Array)
```

Reply:
Fixed, now also works with ranges

Reply:
Thanks! Fixed!

Reply:
I did not get the 
`NoMethodError (undefined method `oop?' for 1..2:Range)`
error.
oop? is a class defined in the helpers.rb and it's being required at the top of the enumerables.rb file so it should be defined